### ISSUE TITLE
Not able publish npm package Failed PUT 403

### ISSUE THREAD

**Issue by aravindqb** [npm/npm#9874]
Hi,
I am not able to publish an npm package using the npm publish command.
I tried logging out and then logging in again, still, its not working. 

0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'publish' ]
2 info using npm@3.3.6
3 info using node@v4.1.2
4 verbose publish [ '.' ]
5 silly cache add args [ '.', null ]
6 verbose cache add spec .
7 silly cache add parsed spec Result {
7 silly cache add   raw: '.',
7 silly cache add   scope: null,
7 silly cache add   name: null,
7 silly cache add   rawSpec: '.',
7 silly cache add   spec: '/home/aravind/demo-app',
7 silly cache add   type: 'directory' }
8 verbose addLocalDirectory /home/aravind/.npm/demo-app/1.0.0/package.tgz not in flight; packing
9 info lifecycle demo-app@1.0.0~prepublish: demo-app@1.0.0
10 silly lifecycle demo-app@1.0.0~prepublish: no script for prepublish, continuing
11 verbose tar pack [ '/home/aravind/.npm/demo-app/1.0.0/package.tgz',
11 verbose tar pack   '/home/aravind/demo-app' ]
12 verbose tarball /home/aravind/.npm/demo-app/1.0.0/package.tgz
13 verbose folder /home/aravind/demo-app
14 verbose addLocalTarball adding from inside cache /home/aravind/.npm/demo-app/1.0.0/package.tgz
15 silly cache afterAdd demo-app@1.0.0
16 verbose afterAdd /home/aravind/.npm/demo-app/1.0.0/package/package.json not in flight; writing
17 verbose afterAdd /home/aravind/.npm/demo-app/1.0.0/package/package.json written
18 silly publish { name: 'demo-app',
18 silly publish   version: '1.0.0',
18 silly publish   readme: 'ERROR: No README data found!',
18 silly publish   _id: 'demo-app@1.0.0',
18 silly publish   scripts: {},
18 silly publish   _shasum: '74e4d81de04d0fe2dff5a62c73ae239045932f60',
18 silly publish   _from: '.' }
19 verbose getPublishConfig undefined
20 silly mapToRegistry name demo-app
21 silly mapToRegistry using default registry
22 silly mapToRegistry registry https://registry.npmjs.org/
23 silly mapToRegistry uri https://registry.npmjs.org/demo-app
24 verbose publish registryBase https://registry.npmjs.org/
25 silly publish uploading /home/aravind/.npm/demo-app/1.0.0/package.tgz
26 verbose request uri https://registry.npmjs.org/demo-app
27 verbose request sending authorization for write operation
28 info attempt registry request try #1 at 3:20:54 PM
29 verbose request using bearer token for auth
30 verbose request id 33c80a78b9da1547
31 http request PUT https://registry.npmjs.org/demo-app
32 http 403 https://registry.npmjs.org/demo-app
33 verbose headers { 'content-type': 'application/json',
33 verbose headers   'cache-control': 'max-age=60',
33 verbose headers   'content-length': '100',
33 verbose headers   'accept-ranges': 'bytes',
33 verbose headers   date: 'Wed, 07 Oct 2015 09:50:55 GMT',
33 verbose headers   via: '1.1 varnish',
33 verbose headers   connection: 'keep-alive',
33 verbose headers   'x-served-by': 'cache-sin6921-SIN',
33 verbose headers   'x-cache': 'MISS',
33 verbose headers   'x-cache-hits': '0',
33 verbose headers   'x-timer': 'S1444211454.307756,VS0,VE854' }
34 verbose request invalidating /home/aravind/.npm/registry.npmjs.org/demo-app on PUT
35 error publish Failed PUT 403
36 verbose stack Error: You do not have permission to publish 'demo-app'. Are you logged in as the correct user? : demo-app
36 verbose stack     at makeError (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:264:12)
36 verbose stack     at CachingRegistryClient.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:252:14)
36 verbose stack     at Request._callback (/usr/local/lib/node_modules/npm/node_modules/npm-registry-client/lib/request.js:172:14)
36 verbose stack     at Request.self.callback (/usr/local/lib/node_modules/npm/node_modules/request/request.js:198:22)
36 verbose stack     at emitTwo (events.js:87:13)
36 verbose stack     at Request.emit (events.js:172:7)
36 verbose stack     at Request.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/request/request.js:1063:14)
36 verbose stack     at emitOne (events.js:82:20)
36 verbose stack     at Request.emit (events.js:169:7)
36 verbose stack     at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/request/request.js:1009:12)
37 verbose statusCode 403
38 verbose pkgid demo-app
39 verbose cwd /home/aravind/demo-app
40 error Linux 3.13.0-65-generic
41 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "publish"
42 error node v4.1.2
43 error npm  v3.3.6
44 error code E403
45 error You do not have permission to publish 'demo-app'. Are you logged in as the correct user? : demo-app
46 error If you need help, you may report this error at:
46 error     <https://github.com/npm/npm/issues>
47 verbose exit [ 1, true ]

**Comment by watilde** [comment]
Hi aravindqb,

The namespace [demo-app](https://www.npmjs.com/package/demo-app) has already taken by another developer, so you don't have permission to publish it to registry. You need to update the name field in your package.json to another name that is not used on the npm registry, or add a scope to name field.
> 36 verbose stack Error: You do not have permission to publish 'demo-app'. Are you logged in as the correct user? : demo-app

See also:
+ https://docs.npmjs.com/files/package.json#name
+ https://docs.npmjs.com/misc/scope

**Comment by aravindqb** [comment]
Thank you @watilde   
Issue solved.

**Comment by jbazi** [comment]
@watilde Thanks. After several attempts of deleting and recreating my json package file and  updating my npm, you saved the day with this simple and quick fix...

**Comment by rasik210** [comment]
TL;DR I simply deleted npm's registry config file from user's directory to start afresh.

My full story:

NPM's registry config file is found at C:\Users\<username>\.npmrc

The problem was that since we use a local npm package repo for controls commonly used in our company so its setting were present already in that file as below:

registry=http://<IP address of NPM local package server of my company>:80/

So whenever I was trying to publish my app to NPM nodeJs server, it was actually trying to contact the NPM local package server of my company instead which was failing.

Then I deleted the file and started all over from npm adduser command. Then it created a fresh entry into registry config file of NPM as shown below:

//registry.npmjs.org/:_authToken=ad3za2pq-x588-4547-8410-3erd25f3t333

From here onwards my publish command started to work as it was now able to contact the npmJs registry server for publishing my new package.

**Comment by watilde** [comment]
@rasik210 Hey it's really nice to share your knowledge, but you shouldn't share your autoToken anyway.

**Comment by rasik210** [comment]
Yes. Very true :). I was absolutely cautious over there. I had changed every single alphabet and number in the authToken before posting. I should have rather used some explanatory text like I did for "\<userName\>" while mentioning NPM's registry config file path.

**Comment by laurci** [comment]
Saved my day man :) wrong config in .npmrc :+1:

**Comment by davidstivcontre1** [comment]
Buenas noches compañeros programadores, ¿como puedo cambiar el nombre si ya tengo todo creado?
